fw/tls.c

```diff
@@ -593,6 +594,15 @@ tfw_tls_send(TlsCtx *tls, struct sg_table *sgt)
 	r = ss_send(conn->cli_conn.sk, &io->skb_list, flags);
 	WARN_ON_ONCE(!(flags & SS_F_KEEP_SKB) && io->skb_list);
 
+out:
+	/*
+	 * TTLS_F_ST_SHUTDOWN and TTLS_F_ST_CLOSE flags are used
+	 * to specify type of closing procedure. We save one of these
+	 * flags previosly in st_flags field, use it in this function
+	 * and reset them in st_flags field to be shure that we can
+	 * use st_flags field for the same purpose.
+	 */
+	io->st_flags &= ~(TTLS_F_ST_SHUTDOWN | TTLS_F_ST_CLOSE);
```

Comment:
These flags are used when we close client connection. They are used when we call `ttls_close_notify` or send  `ttls_send_alert`  with fatal message. For example we try to close connection with TTLS_F_ST_SHUTDOWN flag, because some request was incorrect. This function fails and we don't reset io->st_flags. Later when we handle error of this function we try to close connection immediately with TTLS_F_ST_CLOSE flag, so both flags in st_flags are set and we catch BUG_ON in `tfw_tls_close_msg_flags`.

Reply:
OK, thank you. Please add this either to code comment or a commit message